dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-presto/src/main/java/org/apache/dolphinscheduler/plugin/datasource/presto/param/PrestoDataSourceProcessor.java

```diff
@@ -70,15 +73,16 @@ public BaseDataSourceParamDTO createDatasourceParamDTO(String connectionJson) {
     public BaseConnectionParam createConnectionParams(BaseDataSourceParamDTO datasourceParam) {
         PrestoDataSourceParamDTO prestoParam = (PrestoDataSourceParamDTO) datasourceParam;
         String address = String.format("%s%s:%s", Constants.JDBC_PRESTO, prestoParam.getHost(), prestoParam.getPort());
-        String jdbcUrl = address + "/" + prestoParam.getDatabase();
+        String jdbcUrl = address + Constants.SLASH + prestoParam.getDatabase();
 
         PrestoConnectionParam prestoConnectionParam = new PrestoConnectionParam();
         prestoConnectionParam.setUser(prestoParam.getUserName());
         prestoConnectionParam.setPassword(PasswordUtils.encodePassword(prestoParam.getPassword()));
         prestoConnectionParam.setOther(transformOther(prestoParam.getOther()));
         prestoConnectionParam.setAddress(address);
         prestoConnectionParam.setJdbcUrl(jdbcUrl);
-        prestoConnectionParam.setDatabase(prestoParam.getDatabase());
+        prestoConnectionParam.setDatabase(prestoParam.getDatabase().split(Constants.SLASH)[0]);
+        prestoConnectionParam.setSchema(prestoParam.getDatabase().split(Constants.SLASH)[1]);
```

Comment:
If user set the database not in `database/schema` format, it will error.

Why not add the schema input in the UI for presto datasource?

Reply:
I also thought about modifying the UI before, because I couldn't find a front-end developer in the company to modify it temporarily, but I needed to use the feature at that time, so I changed it in this way.

Reply:
> I also thought about modifying the UI before, because I couldn't find a front-end developer in the company to modify it temporarily, but I needed to use the feature at that time, so I changed it in this way.

@Amy0104 @songjianet do you have time to take a look?